### ISSUE TITLE
FLAML: New feature

### ISSUE THREAD

**Issue by nabalamu** [microsoft/FLAML#97]
My team is working on a multiclass classification model for a predicting the workload type(POC, Prod, Dev, Test) of Azure services like SQLDW, Synapse and SQLDB. We replaced the Gridsearch/XGBoost with FLAML XGBoost for better performance. Since it is multiclass classification, we implemented more metrics like normalized confusion matrix, Precision-Recall curve and Roc-curve using OneVsRestClassifier for binarizing the labels for our final model so that we can measure the performance for prediction of each individual workload type, in additional to accuracy, precision and recall of overall model. This seems like a common requirement that other FLAML users might have and it will be valuable to add these features for multiclass classification models.

The link to access the jupyter notebook for multiclass classification is 
https://microsoft.sharepoint.com/:u:/t/AzureDataUXBA-DataEngineeringandAnalysis/ETY_DWyvPXBEl2S-R5C6rVUBFa0fvbnE9V7KSzAC3H8uMQ?e=hgxKmb

It has the implementation of above metrics in the last section of the file(5. Metrics).

**Comment by sonichi** [comment]
thanks @nabalamu for the feedback. What is a desirable API? Would the following work?

```python
def plot_confusion_matrix(estimator, X_train, y_train, X_test, y_test):
    '''Binarize the data for multi-class tasks and plot confusion matrix

    Args:
        estimator: A multi-class classification estimator
        X_train: A numpy array or a pandas dataframe of training data
        y_train: A numpy array or a pandas series of training labels
        X_test: A numpy array or a pandas dataframe of test data
        y_test: A numpy array or a pandas series of test labels
    '''
```

**Comment by nabalamu** [comment]
Sure, the above API works. Similar APIs can be implemented for ROC curve and Precision-Recall curve as well.

def plot_roc_curve(estimator, X_train, y_train, X_test, y_test):
    '''Binarize the data for multi-class tasks and plot ROC curve

    Args:
        estimator: A multi-class classification estimator
        X_train: A numpy array or a pandas dataframe of training data
        y_train: A numpy array or a pandas series of training labels
        X_test: A numpy array or a pandas dataframe of test data
        y_test: A numpy array or a pandas series of test labels
    '''


def plot_pr_curve(estimator, X_train, y_train, X_test, y_test):
    '''Binarize the data for multi-class tasks and plot Precision-Recall curve

    Args:
        estimator: A multi-class classification estimator
        X_train: A numpy array or a pandas dataframe of training data
        y_train: A numpy array or a pandas series of training labels
        X_test: A numpy array or a pandas dataframe of test data
        y_test: A numpy array or a pandas series of test labels
    '''

**Comment by sonichi** [comment]
There are a few hardcoded things like `figsize=(10,10)` and `fmt='.2f'` in your code. Do you prefer making the plot function outside flaml so that you can customize these visual elements, or keep them hardcoded in flaml?
